### PR TITLE
Add REST API

### LINKED ISSUE #19: APIs
Can the faraday functionality be made available over APIs (REST preferably), to enable integration with UI solutions like RTL?

### PR DESCRIPTION
**Pull request by guggero**
Fixes #19 by adding the same REST proxy (`grpc-gateway`) that is also used in `lnd` and `loopd`.
Also adds a `CORS` option that sets the `Access-Control-Allow-Origin` header if specified.

### CODE REVIEW

File: frdrpc/rpcserver.go [hunk 1]
```diff
@@ -101,9 +132,57 @@ func (s *RPCServer) Start() error {
 
 	}
 	s.rpcListener = grpcListener
+	log.Infof("gRPC server listening on %s", grpcListener.Addr())
 
 	RegisterFaradayServerServer(s.grpcServer, s)
 
+	// We'll also create and start an accompanying proxy to serve clients
+	// through REST. An empty address indicates REST is disabled.
```
[the remark this answers is not in the document]
Comment (guggero) [reply frdrpc/rpcserver.go:140]: Sounds good to me. Removed the default value.

**Comment by carlaKC** [comment]
Linter failing, need to delete `defaultRESTListen` :)

**Comment by guggero** [comment]
Oops, fixed.

**Comment by guggero** [comment]
Sorry, I had a different version of the compiler/gateway because I was working on the update in `lnd`. Should all be fixed now.

**Comment by carlaKC** [comment]
np, all LGTM now! Thanks for a great addition 👾

**Pull request merged**